usaspending_api/common/csv_helpers.py

```diff
@@ -0,0 +1,106 @@
+from django.core.urlresolvers import resolve
+from django.utils.six.moves.urllib.parse import urlparse
+
+from django.conf import settings
+
+import boto3
+import botocore
+import json
+
+
+def s3_get_url(path, checksum):
+    '''
+    Returns a pre-signed S3 URL for the CSV file, or None if the file does not exist
+    '''
+    s3 = boto3.resource('s3')
+    filename = create_filename_from_options(path, checksum)
+
+    bucket = s3.Bucket(settings.CSV_S3_BUCKET)
+    try:
+        obj = bucket.Object(key=filename)
+        if obj:
+            url = s3.generate_presigned_url(
+                ClientMethod='get_object',
+                Params={
+                    'Bucket': settings.CSV_S3_BUCKET,
+                    'Key': filename
+                }
+            )
+
+            return url
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] != "404":
+            raise
+    finally:
+        return None
+
+
+def s3_empty_bucket():
+    '''
+    Deletes all keys in the S3 bucket
+    '''
+    s3 = boto3.resource('s3')
+    bucket = s3.Bucket(settings.CSV_S3_BUCKET)
+    s3.meta.client.head_bucket(Bucket=settings.CSV_S3_BUCKET)
+    for key in bucket.objects.all():
+        key.delete()
+    return True
+
+
+def sqs_add_to_queue(path, checksum):
+    '''
+    Adds a request to generate a CSV file to the SQS queue
+    '''
+    sqs = boto3.client('sqs', region_name=settings.SQS_REGION)
+
+    sqs.send_message(QueueUrl=settings.SQS_QUEUE_URL, MessageBody=json.dumps({
+        "request_checksum": checksum,
+        "request_path": format_path(path),
+    }))
+
+
+def format_path(path):
```

Comment:
This looks like an ideal function for unit-testing - easy to make tests for, and there are a bunch of corner cases that just might trip it up...